model/config.go

```diff
@@ -2430,6 +2430,27 @@ func (o *Config) ToJson() string {
 	return string(b)
 }
 
+func ConfigToJsonWithoutEmptyFields(o *Config) string {
+	if o == nil {
+		return "{}"
+	}
+
+	// remove the nulls, the {}s and the empty strings from the json response. Need to do this multiple times
+	// because we have nested structures
+	re := regexp.MustCompile(`"[^"]+":((null)|(\{\})|("")),?`)
```

Comment:
One other comment: fixing things via regex /seems/ brittle. All of the fields in `model.Config` /should/ be pointers. Also I could imagine a scenario where an empty string for a configuration is the desired required configuration.

I see we're using this in two places:
* to describe the config without enumerating the `nil` fields
* to ingest the `RequiredConfig` from the manifest in the sample plugin

I'll comment on the former here and the latter in the sample plugin PR, but I'm wondering if we 
should just simplify to reporting that the required configuration doesn't match for now, and iterate separately on ways to "describe" that required config. The ideal solution would be to implement a recursive struct walk via the reflect package, and fix all the weird ways we're basically doing that already in other bits of the code. This seems like a good, separate ticket.